packages/contracts/src/PoolMetadataRegistry.sol

```diff
@@ -51,18 +82,40 @@ contract PoolMetadataRegistry is IPoolMetadataRegistry {
     /// @notice Updates the pool metadata CID
     /// @param poolId The pool ID to update the metadata
     /// @param metadataCID The metadataCID related to the new pool metadata
+    function _setPoolMetadata(bytes32 poolId, string memory metadataCID) internal {
+        require(bytes(metadataCID).length > 0, "CID cannot be empty");
+        require(bytes(metadataCID).length <= 46, "CID too long");
+
+        poolIdMetadataCIDMap[poolId] = metadataCID;
+        emit PoolMetadataUpdated(poolId, metadataCID);
+    }
+
+    /// @notice Ensures that the pool exists and the caller is allowed to change its metadata
+    /// @param poolId The pool ID to update the metadata
+    /// @param metadataCID The metadataCID related to the new pool metadata
     function setPoolMetadata(bytes32 poolId, string memory metadataCID)
         public
-        onlyRegisteredPool(poolId)
-        authenticate(poolId)
+        onlyRegisteredPools(_wrapPoolId(poolId))
+        authenticatePools(_wrapPoolId(poolId))
     {
-        poolIdMetadataCIDMap[poolId] = metadataCID;
-        emit PoolMetadataUpdated(poolId, metadataCID);
+        _setPoolMetadata(poolId, metadataCID);
+    }
+
+    function setBatchPoolMetadata(bytes32[] calldata poolIds, string[] calldata metadataCIDs)
```

Comment:
would add documentation for this function as well
https://docs.soliditylang.org/en/v0.8.17/natspec-format.html#tags